shared/language/extract.go

```diff
@@ -1,94 +1,74 @@
 package language
 
 import (
-	"database/sql"
 	"encoding/xml"
+	"errors"
+	"github.com/itsabot/abot/core/log"
+	"github.com/itsabot/abot/shared/datatypes"
+	"github.com/itsabot/abot/shared/helpers/address"
+	"github.com/jmoiron/sqlx"
 	"io/ioutil"
 	"net/http"
 	"net/url"
 	"os"
 	"regexp"
 	"strconv"
 	"strings"
-
-	"github.com/itsabot/abot/shared/datatypes"
-	"github.com/itsabot/abot/shared/helpers/address"
-	"github.com/itsabot/abot/core/log"
-	"github.com/jmoiron/sqlx"
 )
 
 var regexCurrency = regexp.MustCompile(`\d+\.?\d*`)
 var regexNum = regexp.MustCompile(`\d+`)
 var regexNonWords = regexp.MustCompile(`[^\w\s]`)
 
-// ExtractCurrency returns a pointer to a string to allow a user a simple check
-// to see if currency text was found. If the response is nil, no currency was
-// found. This API design also maintains consistency when we want to extract and
-// return a struct (which should be returned as a pointer).
-func ExtractCurrency(s string) sql.NullInt64 {
+var ErrNotFound error = errors.New("Couldn't extract requested type from string")
```

Comment:
Needs a comment since it's exported. Error text should be lowercase, since it's an internal error (i.e. never presented to an end-user).